group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -444,21 +416,21 @@ public CompletableFuture<HeartbeatResponseData> heartbeat(
         return runtime.scheduleReadOperation("classic-group-heartbeat",
             topicPartitionFor(request.groupId()),
             (coordinator, __) -> coordinator.classicGroupHeartbeat(context, request)
-        ).exceptionally(exception -> {
-            if (!(exception instanceof KafkaException)) {
-                log.error("Heartbeat request {} hit an unexpected exception: {}",
-                    request, exception.getMessage());
-            }
-
-            if (exception instanceof CoordinatorLoadInProgressException) {
-                // The group is still loading, so blindly respond
-                return new HeartbeatResponseData()
-                    .setErrorCode(Errors.NONE.code());
+        ).exceptionally(exception -> handleOperationException(
+            "Heartbeat",
```

Comment:
Maybe worth leaving a comment in the helper that some requests further transform errors.

Reply:
Added a not to the javadoc.